### DISCUSSION TITLE
Ideeas on implementing product attributes/swatches

### DISCUSSION THREAD

**Discussion by prodixx** [Laravel-Backpack/community-forum#383] [Q&A (Help)]
So, i need to implement a system for product attributes like: color, size, destination (inside/outside), etc. Some attributes can have only a single value chosen (size, color), and some can have multiple values chosen - destination can be inside and outside at the same time.

Even more, attributes can be added into sets of attributes, like this: 
- set for footwear: color, size, sole
- set for clothes: color, size

I've thought doing it like this: on product create/edit form, generate fields for all attributes and also add a field for choosing the set. Based on what set you choose, hide/show attribute fields with CrudField js library.

Did you had to do such system with backpack? Or maybe do you have a better solution/ idea that you can share?

Thanks.

**Comment by jcastroa87** [comment]
Hello @prodixx 

This sounds like a very specific feature to your project needs, so it would make sense to create a custom solution using JavaScript API.

For the product attributes system, one option is to generate the attribute fields dynamically based on the chosen set. As you mentioned, using the CrudField JavaScript library is a good approach.

Backpack doesn’t have this out-of-the-box since it’s a very customized requirement, but the JavaScript API it offers is quite flexible for implementing this type of dynamic form handling. If you're comfortable with JavaScript, you should be able to implement this by hooking into Backpack’s events.

Let me know if you have further questions, and I’d be happy to help!

Cheers.
